### PR TITLE
fix issue 23607 - invalid code leading to an ICE due to an error mess…

### PR DESCRIPTION
**Pull request by ghost**
…age that's not emitted anymore

https://github.com/dlang/dmd/pull/13946 unfortunately removed an error message that was covered by a single test, which maybe leads the author to think that it was not required anymore.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by SixthDot** [comment]
@thewilsonator, I had to change the fix a bit so merge label must be reset.

File: compiler/src/dmd/typesem.d [hunk 1]
```diff
@@ -317,8 +317,13 @@ private void resolveHelper(TypeQualified mt, const ref Loc loc, Scope* sc, Dsymb
             {
                 if (id.dyncast() == DYNCAST.dsymbol)
                 {
-                    // searchX already handles errors for template instances
```
Comment (ibuclaw) [inline compiler/src/dmd/typesem.d:320]: This raises some questions about searchX here. Is there something that function isn't doing?

File: compiler/src/dmd/typesem.d [hunk 2]
```diff
@@ -317,8 +317,13 @@ private void resolveHelper(TypeQualified mt, const ref Loc loc, Scope* sc, Dsymb
...
-                    assert(global.errors);
+                    auto ti = (cast(Dsymbol)id).isTemplateInstance();
+                    assert(ti);
+                    sm = s.search_correct(ti.name);
+                    if (sm)
+                        .error(loc, "template identifier `%s` is not a member of %s `%s`, did you mean %s `%s`?", ti.name.toChars(), s.kind(), s.toPrettyChars(), sm.kind(), sm.toChars());
```
Comment (dkorpel) [inline compiler/src/dmd/typesem.d:324]: Can you add this branch to the test as well?

File: compiler/src/dmd/typesem.d (hunk 1 above)
Comment (SixthDot) [reply compiler/src/dmd/typesem.d:320]: isnt doing "anymore" ? yes. See the diff of #13946.

**Pull request closed without merging**